tortoisebeacon/weakcoin/weak_coin.go

```diff
@@ -206,34 +239,55 @@ func (wc *weakCoin) calculateWeakCoin(epoch types.EpochID, round types.RoundID)
 	wc.weakCoinsMu.Lock()
 	defer wc.weakCoinsMu.Unlock()
 
-	if len(wc.proposals[pair]) == 0 {
-		wc.Log.Warning("No weak coin proposals were received",
+	proposals := wc.proposals[pair]
+
+	if len(proposals) == 0 {
+		wc.Log.With().Warning("No weak coin proposals were received",
 			log.Uint64("epoch_id", uint64(epoch)),
-			log.Uint64("round", uint64(round)))
+			log.Uint64("round_id", uint64(round)))
 
 		wc.weakCoins[pair] = false
 
 		return
 	}
 
-	smallest := new(big.Int).SetBytes(wc.proposals[pair][0][:])
+	smallest := new(big.Int).SetBytes(proposals[0])
+	stringProposals := []string{smallest.String()}
 
-	for i := 1; i < len(wc.proposals[pair]); i++ {
-		current := new(big.Int).SetBytes(wc.proposals[pair][i][:])
+	for i := 1; i < len(proposals); i++ {
+		current := new(big.Int).SetBytes(proposals[i])
+		stringProposals = append(stringProposals, current.String())
 
 		if current.Cmp(smallest) == -1 {
 			smallest = current
 		}
 	}
 
-	result := &big.Int{}
-	result.And(smallest, big.NewInt(1))
+	coinBit := new(big.Int).And(smallest, big.NewInt(1))
```

Comment:
smallest.Bit(0)

Reply:
Indeed. I didn't think of it. Looks way better